perfkitbenchmarker/benchmark_spec.py

```diff
@@ -309,12 +314,15 @@ def PrepareVm(self, vm, firewall):
       firewall.AllowPort(vm, port)
     vm.AddMetadata(benchmark=self.benchmark_name)
     vm.WaitForBootCompletion()
-    vm.OnStartup()
     if FLAGS.scratch_disk_type == disk.LOCAL:
       vm.SetupLocalDisks()
     for disk_spec in vm.disk_specs:
       vm.CreateScratchDisk(disk_spec)
 
+    # This must come after Scratch Disk creation to support the
+    # Containerized VM case
+    vm.OnStartup()
```

Comment:
In that case, is moving the package-manager specific commands (i.e. AptUpdate) to a different method (maybe PrepareVMEnvironment?) that gets called after Scratch disk creation a viable idea? docker run has to be called after mounting the volumes, but it has to be called before any apt-get commands